work/v1/types.go

```diff
@@ -66,6 +70,64 @@ type DeleteOption struct {
 	SelectivelyOrphan *SelectivelyOrphan `json:"selectivelyOrphans,omitempty"`
 }
 
+type StatusFeedbackOption struct {
+	// ResourceIdentifier represents the group, resource, name and namespace of a resoure.
+	// +kubebuilder:validation:Required
+	// +required
+	ResourceIdentifier ResourceIdentifier `json:"resourceIdentifier"`
+
+	// FeedbackRules defines what resource status field should be returned.
+	// +kubebuilder:validation:Required
+	// +required
+	FeedbackRules []FeedbackRule `json:"feedbackRule"`
+}
+
+type FeedbackRule struct {
+	// Type defines the option of how status can be returned.
+	// It can be jsonPaths only or wellKnownStatus.
+	// If the type is jsonPaths, user should specify the jsonPaths field
+	// If the type is wellKnownStatus, certain commont fields of status will be reported,
+	// including replicas, readyReplicas, and availableReplicas. And if these status fields
+	// do not exist, no values will be reported.
+	// +kubebuilder:validation:Required
+	// +required
+	Type FeedBackType `json:"type"`
+
+	// JsonPaths defines the json path under status field to be synced.
+	// +optional
+	JsonPaths []JsonPath `json:"jsonPaths,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=wellKnownStatus;jsonPaths
+type FeedBackType string
+
+const (
+	// WellKnownStatusType represents that values of some common status fields will be returned, which
+	// include replica, readyReplica and availableReplica.
+	WellKnownStatusType FeedBackType = "wellKnownStatus"
+
+	// JsonPathsType represents that values of status fields with certain json paths specified will be
+	// returned
+	JsonPathsType FeedBackType = "jsonPaths"
+)
+
+type JsonPath struct {
+	// name represents the aliase name for this field
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+	// JsonPaths represents the json path of the field under status.
+	// The path must point to a field with single value in the type of integer, bool or string.
+	// If the path points points to a non-existing field, no value will be returned.
```

Comment:
s/points points/points/

Reply:
when the path points to a non-existing path, a `null` value will be returned instead